### PR TITLE
[plugin.video.massengeschmack] 2.4.1

### PR DESCRIPTION
**Pull request by phoerious**
### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: plugin.video.massengeschmack/resources/lib/listing.py [hunk 1]
```diff
@@ -56,19 +56,19 @@ def show(self):
         if 'true' == ADDON.getSetting('advanced.viewmodeFix'):
             self.setViewMode(510)
         else:
-            self.setViewMode(55)
+            self.setViewMode(self.__source.getViewMode())
```
Comment (enen92) [inline plugin.video.massengeschmack/resources/lib/listing.py]: Sorry this is not allowed according to our rules (https://kodi.wiki/view/Add-on_rules). I am not sure how it got unnoticed before but plugins should not force viewtypes on the skin. They should set the mediatype on the listitem and contenttype via xbmcplugin. Views are provided by the skin to each specific media type, the user is then able to change them to their taste.

Comment (phoerious) [reply plugin.video.massengeschmack/resources/lib/listing.py]: Would it be okay if I added a setting so the user can turn off this behaviour? Not having correctly set view modes is really annoying, but if the user wants, they can still turn it off (BTW the add-on *does* set the content type, but it's not enough to get nice-looking and consistent listings).

Comment (MartijnKaijser) [reply plugin.video.massengeschmack/resources/lib/listing.py]: it's pointless cause each skin has different IDs.

Comment (phoerious) [reply plugin.video.massengeschmack/resources/lib/listing.py]: I know. But at least for Estuary it looks decent. For every other skin you can turn it off. For Aeon Nox there is already a setting.

Comment (phoerious) [reply plugin.video.massengeschmack/resources/lib/listing.py]: Yes? No? Using this add-on without consistent view modes is really ugly and confusing.

Comment (enen92) [reply plugin.video.massengeschmack/resources/lib/listing.py]: @phoerious sorry for the delay. Can you make it as a setting (that can be disabled) and only set the viewmode if the user is using estuary? You can detect the skin via `xbmc.getSkinDir()`. Thanks

Comment (phoerious) [reply plugin.video.massengeschmack/resources/lib/listing.py]: I added a setting. This got closed somehow, so here's the new PR: #1949

**Pull request closed without merging**